### PR TITLE
Beta.8

### LINKED ISSUE #585: 2.0.0 Enhancements
- [x] Better injected wallet management, add default wallet images and names
until dynamic packages are loaded)
- [x] Add artificial loader while explorer is opening when there is cached data (animation performance improvement)
- [x] Add hover effect for modal header buttons (need to discuss with Maciej)
- [x] Fix edge case and re-generate wallet connect code on cancel / expiry
- [x] ~Create custom jsonrpc providers for unsupported chains if they provide rpc url in config~ this is now handled on wagmi side
- [x] Add more horizontal space to wallet-button label
- [x] Add app details to config so it shows up nicer in wallets
- [x] Correct mobile re-direct handling for chain switching (why does it deep link after first time instead of universal?)
- [x] Check for issues with reconnection on mobile when safari is fully closed (I suspect we should delay rendering of app 
- [x] Look into react hooks / ethereal api implementation, check if they are unsubscribed correctly
- [x] ~Create `getWeb3Provider` helper and `useWeb3Provider` hook (https://github.com/WalletConnect/web3modal/issues/602)~
- [x] Add ethers as peerDependency
- [x] Fix "…" color in explorer screen
- [x] Move wage/core management to ethereal package, expose its options
- [x] Add buffer polyfill by default once we manage @wagmi/core
- [x] Make higher breakpoint for mobile
- [x] Add modal shadows
- [x] More mobile breakpoints (stretch + add spacing latter)
- [x] Dynamic spacing between wallet reel on mobile to ensure there is indication of scroll
- [x] Play with spring animations

### LINKED ISSUE #663: Sample implementation not working in safari 14
**Describe the bug**
Sample implementation not working in safari. 

**To Reproduce**
1. Go to safari and open  'https://web3modal.com/try-it-out'
2. Click on 'Connect  Wallet'
3. Click on any wallet option
4. See nothing showing there

**Expected behavior**
Need to show the option details

**Screenshots**
Screenshot 2022-10-30 at 9 33 32 AM

**Desktop (please complete the following information):**
 - Browser safari

### LINKED ISSUE #694: ModalCtrl.open() did not return error from Trust Wallet in-app browser
**Describe the bug**
When using single chain as provided in ClientCtrl.setEthereumClient, ModalCtrl.open() will automatically switch to the correct network.  In Trust Wallet in-app browser however, switch network is not supported thus the wallet will return an error but on dApp web, ModalCtrl.open() did not able to catch this error even after surround with try catch. 

**SDK Version**
 - Declared @web3modal/core /ethereum /ui in package.json as ^2.0.0-beta.3. Did npm install to avoid using old version too.

**To Reproduce**
1. Configure chains to only use single chain:
```
    ClientCtrl.setEthereumClient({
      appName: 'appName',
      autoConnect: true,
      chains: environment.production ? [chains.binanceSmartChain] : [chains.binanceSmartChainTestnet],
      providers: [providers.walletConnectProvider({ projectId: environment.walletConnectProjectId })],
    })
```
2. Set wallet to use different network from declared in chains above.
3. Proceed to connect wallet in dApp using Trust Wallet option in Web3Modal.
4. See error generated in console.

**Expected behavior**
- Surrounded ModalCtrl.open() with try catch should be able to catch the error.
- or ModalCtrl.open() can return Promise<boolean> to indicate of connection successful/unsuccessful.

**Screenshots**
image

**Smartphone (please complete the following information):**
 - Device: Xiaomi Mi 9T
 - OS: Android 10
 - Browser Trust Wallet In-app browser
 - Version Trust Wallet 6.55

**Additional context**
Works correctly on wallet that supports switch network - MetaMask extension, MetaMask in-app browser.

### LINKED ISSUE #719: Account modal view
- [ ] ENS Resolution
- [ ] Balance
- [ ] Realtime flag?
- [ ] Manual recent transactions?
- [ ] hook / action to open account modal

### LINKED ISSUE #741: The imToken mobile application does not launch
**Describe the bug**
I installed the imToken app, and as I press on it, the app (on mobile) does not launch...

The only visible log in safari is:

```
[Error] Unhandled Promise Rejection: TypeError: undefined is not an object (evaluating 't.renderRoot.querySelector')
	dispatchException (622:2235)
	(fonction anonyme) (622:124)
	asyncGeneratorStep (10:4)
	_next (10:23)
	(fonction anonyme) (10:28)
	Promise
	(fonction anonyme) (10:20)
	(fonction anonyme) (929:2469)
	(fonction anonyme) (1366:261)
	promiseReactionJob
```

**SDK Version**
 - Version [2.0.0 BETA 7]

**Screenshots**
imtoken

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone 13 Pro]
 - OS: [e.g. iOS 16.1]
 - Browser [Safari]

### LINKED ISSUE #749: Empty network / wallet view on iOS 15.2 using Chrome Browser
**Describe the bug**
https://twitter.com/wolfram_eth/status/1599367190615990272
User is having issues on ios 15.2 when using chrome browser (network view is empty)

**SDK Version**
 - Version beta.7 - beta.8

**To Reproduce**
Need to simulate ios 15 version (likely 14 as well) and text out the modal on chrome browser.

**Expected behavior**
Everything should work fine

### PR DESCRIPTION
**Pull request by IljaDaderko**
- [x] Minimal implementation for account view
- [ ] Document missing props / web-component attributes for Web3Button component
- [x] Build and tag beta.8
- [ ] Update web3modal.com once released
- [x] Address avatar component - generates gradient based on address
- [x] Extend OptionsCtrl with account and network state and actions
- [x] Remove background accent color as per figma updates
- [x] Cleanup and unify layouts, use grids where it makes sense
- [x] Safeguard against universal link passed as deepling and vice versa
- [x] Sow error toast and re-generate qrcode if user rejects connection
- [x] ResizeObserver and svg mask fixes for iOS 15/14 
- [x] Fix loader displayed on subsequent visit to explorer page when all wallets are loaded already
- [x] Network switching flow when connected

close #741 
close #749
close #694
close #585
close #663
close #586

### CODE REVIEW

**Comment by IljaDaderko** [comment]
@WarrickFitz I updated this pr to account for your fix. Slightly different approach, but it should work. It will redirect to http url and append `/wc?uri=${encodedWcUrl}`

I also added your wallet to custom example, you can play with it here https://beta-8.web3modal-dev.pages.dev/custom (under View More), as far as I can tell it redirects correctly (but to 404 page).

Also, desktop wallets will still prioritise deeplinks first which don't open as `_blank` target, so unless you have an actual deepling to native app, set it to empty string or null, so that it is not used

**Comment by WarrickFitz** [comment]
Thanks so much @IljaDaderko - looks great. I had not considered setting the native to an empty string, but makes a lot of sense.

File: chains/ethereum/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3modal/ethereum",
-  "version": "2.0.0-beta.7",
+  "version": "2.0.0-b648e83",
```
Comment (crypblizz8) [inline chains/ethereum/package.json:3]: Meant to be 0.8 ?

File: examples/react-standalone/package.json [hunk 2]
```diff
@@ -1,21 +1,21 @@
 {
   "name": "react-standalone-example",
-  "version": "2.0.0-beta.7",
+  "version": "2.0.0-b648e83",
```
Comment (crypblizz8) [inline examples/react-standalone/package.json:3]: Similar change here? Or I might be missing something?

File: chains/ethereum/package.json (hunk 1 above)
Comment (IljaDaderko) [reply chains/ethereum/package.json:3]: Will tag before merging 👍 this is for canary

File: examples/react/src/pages/_app.tsx [hunk 3]
```diff
@@ -27,12 +28,19 @@ export const ethereumClient = new EthereumClient(wagmiClient, chains)
...
+  useEffect(() => {
+    setReady(true)
+  }, [])
 
   return (
     <>
-      <WagmiConfig client={wagmiClient}>
-        <Component {...pageProps} />
-      </WagmiConfig>
+      {ready ? (
```
[the remark this answers is not in the document]
Comment (IljaDaderko) [reply examples/react/src/pages/_app.tsx:39]: Causes hydration errors in next js as different data is returned when user is connected on the server and client. Wagmi has this same approach in their templates.

File: packages/ui/src/partials/w3m-avatar/index.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  // -- render ------------------------------------------------------- //
+  protected render() {
+    const classes = {
+      'w3m-avatar': true,
+      'w3m-avatar-small': this.size === 'small',
+      'w3m-avatar-medium': this.size === 'medium'
+    }
+
+    if (this.address) {
+      UiUtil.generateAvatarColors(this.address)
```
Comment (crypblizz8) [inline packages/ui/src/partials/w3m-avatar/index.ts:42]: clean + light weight ✅

File: packages/ui/src/partials/w3m-desktop-wallet-selection/styles.css [hunk 5]
```diff
@@ -1,23 +1,24 @@
-.w3m-view-row {
-  display: flex;
-  width: 100%;
+.w3m-grid {
+  display: grid;
+  grid-template-columns: repeat(4, 80px);
```
Comment (crypblizz8) [inline packages/ui/src/partials/w3m-desktop-wallet-selection/styles.css:3]: NoAction Needed: Cool syntax, first time seeing it.

**Review by crypblizz8 — DISMISSED** [review]
LGTM, great refactor + lightweight + account feature 🙏

File: packages/ui/src/components/w3m-network-image/index.ts [hunk 6]
```diff
@@ -1,24 +1,45 @@
...
-      </div>
-    `
+    const src = UiUtil.getChainIcon(this.chainId)
+
+    if (src) {
+      return html`
+        <div class="w3m-network-image">
+          <svg width="54" height="59" viewBox="0 0 54 59" fill="none">
+            <defs>
+              <clipPath id="polygon">
```
[the remark this answers is not in the document]
Comment (crypblizz8) [reply packages/ui/src/components/w3m-network-image/index.ts:24]: 🤦 on my end

**Pull request merged**